piecedirectory/piecedirectory_test.go

```diff
@@ -248,6 +253,61 @@ func testImportedIndex(ctx context.Context, t *testing.T, cl *client.Store) {
 	require.Equal(t, len(blk.RawData()), sz)
 }
 
+func testDataSegmentIndex(ctx context.Context, t *testing.T, cl *client.Store) {
+	// Any calls to get a reader over data should return a reader over the fixture
+	pr := CreateMockPieceReaderFromPath(t, "./testdata/deal.data")
+	fstat, err := os.Stat("./testdata/deal.data")
+	require.NoError(t, err)
+
+	rdr, err := pr.GetReader(ctx, 0, 0, abi.PaddedPieceSize(fstat.Size()))
+	require.NoError(t, err)
+	pieceCid := CalculateCommp(t, rdr).PieceCID
+
+	recs, err := parseShardWithDataSegmentIndex(ctx, pieceCid, fstat.Size(), rdr)
+	require.NoError(t, err)
+	err = cl.AddIndex(ctx, pieceCid, recs, false)
+	require.NoError(t, err)
+
+	// Add deal info for the piece - it doesn't matter what it is, the piece
+	// just needs to have at least one deal associated with it
+	di := model.DealInfo{
+		DealUuid:    uuid.New().String(),
+		ChainDealID: 1,
+		SectorID:    2,
+		PieceOffset: 0,
+		PieceLength: 0,
+	}
+	err = cl.AddDealForPiece(ctx, pieceCid, di)
+	require.NoError(t, err)
+
+	// Remove size information from the index records.
+	// This is to simulate what happens when an index is imported from the
+	// DAG store: only the offset information is imported (not size
+	// information)
+	for i, r := range recs {
+		recs[i] = model.Record{
+			Cid: r.Cid,
+			OffsetSize: model.OffsetSize{
+				Offset: r.Offset,
+				Size:   0,
+			},
+		}
+	}
+
+	// Add the index to the local index directory, marked as incomplete
+	err = cl.AddIndex(ctx, pieceCid, recs, false)
+	require.NoError(t, err)
```

Comment:
I don't think we need to test importing from the dagstore here. Just the data segment index functionality.

I'd suggest we also skip lines 266 - 269. Parsing the piece data and adding the index should all happen as part of AddDealForPiece.